src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -0,0 +1,167 @@
+        if (this.status.state) delete this.status.state;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onMangaClicked(e) {
+        // Hentai check
+        if(e.detail.connector.tags.includes('hentai') || e.detail.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Browsing ' + e.detail.connector.label;
+        this.status['state'] = 'Looking at ' + e.detail.title;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onChapterGetPages(e) {
+        // Hentai check
+        if(e.detail.manga.connector.tags.includes('hentai') || e.detail.manga.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Reading ' + e.detail.manga.title;
+        this.status['state'] = e.detail.title;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    async updateStatus() {
+        if(this.rpc) {
+            if (this.enabled && this.statusNew) {
+                this.IpcBytes = this.rpc.transport.socket.bytesWritten;
+                if( !this.hentai || (this.hentai && this.enabledHentai)) {
+                    this.rpc.setActivity(this.status);
+                } else {
+                    this.statusNew = false;
+                }
+            }
+
+            // Test if IPC is still active
+            if ( this.statusNew && this.rpc.transport.socket.bytesWritten > this.IpcBytes) {
+                this.IpcBytes = this.rpc.transport.socket.bytesWritten;
+                this.statusNew = false;
+            } else if (this.rpc.transport.socket.bytesWritten == this.IpcBytes && this.statusNew) {
+                console.warn('WARNING: DiscordPresence - Lost connection to Discord.');
+                this.stopDiscordPresence();
+            }
+        }
+    }
+
+    stopDiscordPresence() {
+        this.statusNew = false;
+        if (this.rpc) {
+            this.rpc.clearActivity();
+            this.rpc.destroy();
+            this.rpc = null;
+        }
+    }
+
+    async startDiscordPresence() {
+        if (!this.rpc) {
+            this.rpc = new DiscordRPC.Client({ transport: 'ipc' });
+            this.rpc.on('ready', () => {
+                this.status.startTimestamp = + new Date();
+                
+                // some delay for Discord to be receptive
+                setTimeout( () => {
+                    this.updateStatus();
+                }, 2000);
+
+                // activity can only be set every 15 seconds
+                setInterval(() => {
```

Comment:
Reference should be stored in class field, so it can be stopped gracefulley

Reply:
I thought the interval is attached to `this.rpc`. Can you please point me in the right direction?

Reply:
Introduce a field e.g. `this.statusInterval = null;`
Then assign it in the method `this.statusInterval = setInterval(this.updateStatus.bind(this, 15200));`
To stop interval `this.clearInterval(this.statusInterval);`